borg/xattr.py

```diff
@@ -46,6 +46,14 @@ def get_all(path, follow_symlinks=True):
         logger.error(msg)
         raise Exception(msg)
 
+# If we are running with fakeroot on Linux, then use the xattr functions of fakeroot. This is needed by
+# the 'test_extract_capabilities' test, but also allows xattrs to work with fakeroot on Linux in normal use.
+# TODO: Check whether fakeroot supports xattrs on all platforms supported below.
+# TODO: If that's the case then we can make Borg fakeroot-xattr-compatible on these as well.
+LD_PRELOAD = os.environ.get('LD_PRELOAD', '')
+if sys.platform.startswith('linux') and 'fakeroot' in LD_PRELOAD:
+    libc_name = LD_PRELOAD
```

Comment:
ok, so fakeroot emulates libc by using LD_PRELOAD.

but does everything that is loaded by LD_PRELOAD also emulate libc (or at least the stuff we need in this python module)?